library/src/main/java/com/bumptech/glide/request/target/CustomViewTarget.java

```diff
@@ -300,7 +302,8 @@ private void maybeRemoveAttachStateListener() {
     @Nullable
     static Integer maxDisplayLength;
     private final View view;
-    private final List<SizeReadyCallback> cbs = new ArrayList<>();
+    private final Set<SizeReadyCallback> cbs =
```

Comment:
There's typically only one callback (unless you're using lots of thumbnails) and at most there's one callback per request you pass in. It's unlikely that that Set will perform any better than list and it allocates at least an extra object per Entry that's avoided with a list.

Do you see significant gains with these changes?